terraform/modules/configs/deploy-all/main.tf

```diff
@@ -422,6 +450,18 @@ module "load_balancer_client" {
   hosted_zone_name      = data.aws_ssm_parameter.hosted_zone_name_alb_bat_client.value
 }
 
+# I am not sure if this app S3_VIRUS_SCAN app needs a load_balancer_s3_virus_scan
```

Comment:
I think we should route requests from the Lambda through the existing internal NLB, similar to the communication between the BaT client and Spree (rather than create an additional load balancer).  
To do this, you'll need to register the ECS service with a target group & listener in the same way as is done for Spree - see:

https://github.com/Crown-Commercial-Service/ccs-scale-infra-services-bat/blob/develop/terraform/modules/services/spree/ecs.tf#L59
https://github.com/Crown-Commercial-Service/ccs-scale-infra-services-bat/blob/develop/terraform/modules/services/spree/ecs.tf#L69
https://github.com/Crown-Commercial-Service/ccs-scale-infra-services-bat/blob/develop/terraform/modules/services/spree/ecs.tf#L149

Then inject the internal NLB address / listener port into the Lambda function (if I've understood the architecture correctly) 👍